plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -2135,19 +2140,24 @@ function ( $acc, $product ) {
 				);
 			},
 			function ( $collection_args, $query ) {
-				$product_reference = isset( $query['productReference'] ) ? array( $query['productReference'] ) : null;
+				$product_references = isset( $query['productReference'] ) ? array( $query['productReference'] ) : null;
 				// Infer the product reference from the location if an explicit product is not set.
-				if ( empty( $product_reference ) ) {
+				if ( empty( $product_references ) ) {
 					$location = $collection_args['productCollectionLocation'];
 					if ( isset( $location['type'] ) && 'product' === $location['type'] ) {
-						$product_reference = array( $location['sourceData']['productId'] );
+						$product_references = array( $location['sourceData']['productId'] );
 					}
+
 					if ( isset( $location['type'] ) && 'cart' === $location['type'] ) {
-						$product_reference = $location['sourceData']['productIds'];
+						$product_references = $location['sourceData']['productIds'];
+					}
+
+					if ( isset( $location['type'] ) && 'order' === $location['type'] ) {
+						$product_references = $this->get_product_ids_from_order( $location['sourceData']['orderId'] ?? null );
```

Comment:
Not strictly related to this PR:
Might be best to default to `array()` instead of `null`. Both options could cause the variable to change its type during execution.
```suggestion
						$product_references = $this->get_product_ids_from_order( $location['sourceData']['orderId'] ?? array() );
```
Same applies for L2143.

Reply:
I am not sure if I understand the issue correctly. I already updated it to fallback to 0 as suggested in [this comment](https://github.com/woocommerce/woocommerce/pull/52258#discussion_r1815151545). Also, this `null` fallback is for argument of `get_product_ids_from_order()` function and not for `product_references` variable. Am I missing something?

Reply:
No worries! I noticed a refactor task on the roadmap, so since this is just a minor point, I'll go ahead and resolve the conversation here.

As a general rule, especially after PHP 8, changing a variable's type is considered an antipattern. This will also cause a fatal error if the `strict_types` directive is included. Just keep this in mind!